src/FileIO.chpl

```diff
@@ -85,4 +157,33 @@ module FileIO {
         }
         return t;
     }
+
+    proc domain_intersection(d1: domain(1), d2: domain(1)) {
+        var low = max(d1.low, d2.low);
+        var high = min(d1.high, d2.high);
+        if (d1.stride !=1) && (d2.stride != 1) {
+            //TODO: change this to throw
+            halt("At least one domain must have stride 1");
+        }
+        var stride = max(d1.stride, d2.stride);
+        return {low..high by stride};
+    }
+    
+    class ArkoudaFileWriter {
+
+        proc writeDistArrayToDisk(filename: string, dsetname, A) throws {
+
+        }
+
+    }
+
+    class FileWriterRegistry {
```

Comment:
Reading from the PR description, my understanding here is that this is essentially another command map, but I don't know if I understand the need.

Is there a reason that we wouldn't just use the existing command map? It seems to me to be doing the same thing.

Reply:
This kind of goes with your question on `GenHDF5IO` and whether or not it needs to be included in `arkouda_server`.

We need to wire the writing pieces together in some way.  Right now I think this is done down in GenSymIO, so we need some level of indirection.  The `ArkoudaFileWriter` class would act as an interface in this capacity so GenSymIO could be made to function completely off of that class.

We could wire in specific commands (i.e. writeToHDF, writeToParquet, etc.) but depending how far we want to go with I/O abstraction we could have a single proc `writeToFile` and based on a format type passed in, we could pull out a compatible writer (same for reading, maybe base it on file magic numbers).  This is what I was thinking about with a separate writer registry.  I was worried about polluting the cmd map at the expense of extra complexity.